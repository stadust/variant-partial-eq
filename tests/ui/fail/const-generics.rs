use std::marker::PhantomData;
use variant_partial_eq::VariantPartialEq;

#[derive(VariantPartialEq)]
struct MyStruct<const N: u8> {
    ghost: PhantomData<N>
}

fn main() {}