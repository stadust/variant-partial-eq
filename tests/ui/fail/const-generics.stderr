error: #[derive(VariantPartialEq)] does not support const generics
 --> tests/ui/fail/const-generics.rs:5:17
  |
5 | struct MyStruct<const N: u8> {
  |                 ^^^^^

error[E0747]: constant provided when a type was expected
 --> tests/ui/fail/const-generics.rs:6:24
  |
6 |     ghost: PhantomData<N>
  |                        ^
